sensorimotor/sensorimotor/TMRegion.py

```diff
@@ -0,0 +1,571 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2015, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import copy
+from operator import mul
+import numpy
+
+from nupic.bindings.math import GetNTAReal
+from nupic.support import getArgumentDescriptions
+from nupic.regions.PyRegion import PyRegion
+
+from sensorimotor.general_temporal_memory import GeneralTemporalMemory
+from sensorimotor.fast_general_temporal_memory import FastGeneralTemporalMemory
+from nupic.research.monitor_mixin.temporal_memory_monitor_mixin import (
+  TemporalMemoryMonitorMixin)
+
+
+
+class MonitoredFastGeneralTemporalMemory(TemporalMemoryMonitorMixin,
```

Comment:
That might be overly complicated. To be consistent with `TPRegion`, and since all the `TemporalMemory` subclasses have the same interface, we can just have this one work for all of them.